src/hardware/covox.h

```diff
@@ -0,0 +1,45 @@
+/*
+ *  Copyright (C) 2022-2022  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#ifndef DOSBOX_Covox_H
+#define DOSBOX_Covox_H
+
+#include "dosbox.h"
+
+#include "inout.h"
+#include "lpt_dac.h"
+#include "mixer.h"
+
+class Covox final : public LptDac {
+public:
+	Covox() : LptDac("COVOX", ChannelFeature::Mono, covox_uses_mixer_rate) {}
+	void BindToPort(const io_port_t lpt_port) final;
+	void ConfigureFilters(const FilterState state) final;
+
+protected:
+	AudioFrame Render() final;
+
+private:
+	void WriteData(const io_port_t, const io_val_t value, const io_width_t);
+	uint8_t ReadStatus(const io_port_t, const io_width_t);
+	void WriteControl(const io_port_t, const io_val_t, const io_width_t){}
+
+	static constexpr uint16_t covox_uses_mixer_rate = 0u; // 0 means use the mixer
```

Comment:
Why static? A const is a const, no?

Reply:
Yes to both; will move to the mixer. 

Re: static constexpr members, some compilers will fail to find non-static constexpr members in member functions. Const member can be assigned through all the constructor pathways.
Take a member like `const into var = 10;`, it can be changed during construction (and isn't compile time evaluated). Where as the constexpr is baked in and truly immutable. 

None the less.. needing static is quite annoying!:facepalm:

Reply:
C++ never fails to disappoint when it comes to arcane, unmanageable complexity of the core language itself :D